DQ_Kinematics_pose_jacobian_derivative_tests.py

```diff
@@ -0,0 +1,173 @@
+"""(C) Copyright 2022 DQ Robotics Developers
+
+This file is part of DQ Robotics.
+
+    DQ Robotics is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    DQ Robotics is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with DQ Robotics.  If not, see <http://www.gnu.org/licenses/>.
+
+Contributors:
+ - Juan Jose Quiroz Omana -  juanjqo@g.ecc.u-tokyo.ac.jp
+"""
+import unittest
+from dqrobotics import*
+from dqrobotics.robot_modeling import DQ_SerialManipulatorDH
+from dqrobotics.robot_modeling import DQ_SerialManipulatorMDH
+from dqrobotics.robot_modeling import DQ_HolonomicBase
+from dqrobotics.robot_modeling import DQ_DifferentialDriveRobot
+from dqrobotics.robot_modeling import DQ_SerialManipulatorDenso
```

Comment:
Is there a particular reason for this (and other) unused imports?

Reply:
No, I forgot to remove those in which the pose jacobian derivative method is not available yet (DQ_SerialManipulatorDenso, DQ_SerialWholeBody, DQ_WholeBody). I did it now.

Reply:
@juanjqo I would suggest adding a test to check if calling the methods in those classes is generating the correct exception. For example, calling a method that does not exist generates an exception in Python but we have to check in this case if the correct `runtime_error` is being thrown.
https://stackoverflow.com/questions/23337471/how-to-properly-assert-that-an-exception-gets-raised-in-pytest

The reasoning is that, otherwise, some errors in the CPP side can slip away unchecked, similar to the linkage errors described in https://github.com/dqrobotics/cpp/issues/50.

Reply:
@mmmarinho Thank you for your feedback. I completely agree. I implemented theses tests. 
As a matter of fact, I spotted another bug in the class DQ Whole Body by doing so.

PS:
Example of one of the tests recently added:
```Python
def test_whole_body_pose_jacobian_derivative(self):
    njoints = whole_body_robot.get_dim_configuration_space()
    with self.assertRaises(Exception):
      J_dot = whole_body_robot.pose_jacobian_derivative(np.zeros(njoints), np.zeros(njoints))
```
I'm not sure if I implemented the test as you suggested. Please let me know if additional changes are required.